controllers/host/platformnetworks.go

```diff
@@ -5,72 +5,911 @@ package host
 	utils "github.com/wind-river/cloud-platform-deployment-manager/common"
+	"github.com/wind-river/cloud-platform-deployment-manager/controllers/common"
+	cloudManager "github.com/wind-river/cloud-platform-deployment-manager/controllers/manager"
 	v1info "github.com/wind-river/cloud-platform-deployment-manager/platform"
 	"k8s.io/apimachinery/pkg/types"
 	kubeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
 )
 
-// TODO(sriram-gn): All platform network reconciliation workflows will be implemented here.
-// Currently we are just setting status of Reconciled / InSync of every platform network
-// and address pool instances to true just to acknowledge these objects are accepted.
-func (r *HostReconciler) ReconcilePlatformNetworks(client *gophercloud.ServiceClient, instance *starlingxv1.Host, profile *starlingxv1.HostProfileSpec, host *v1info.HostInfo) error {
-	if !utils.IsReconcilerEnabled(utils.HostPlatformNetwork) {
-		return nil
+// makeRangeArray converts an array of range structs to an array of arrays where
+// the inner array contains two elements.  The first element is the range start
+// address and the second element is the range end address.  This is to align
+// with the system API formatting which represents a pair as an array of two
+// elements.
+func makeRangeArray(ranges []starlingxv1.AllocationRange) [][]string {
+	result := make([][]string, len(ranges))
+	for index, r := range ranges {
+		result[index] = []string{r.Start, r.End}
 	}
 
-	addrpools := &starlingxv1.AddressPoolList{}
-	platform_networks := &starlingxv1.PlatformNetworkList{}
-	opts := kubeclient.ListOptions{}
-	opts.Namespace = instance.Namespace
-	err := r.List(context.TODO(), addrpools, &opts)
+	return result
+}
+
+// compareRangeArrays compares two range arrays and returns true if they are
+// equal.
+func compareRangeArrays(x, y [][]string) bool {
+	if len(x) != len(y) {
+		return false
+	}
+
+	count := 0
+	for _, o := range x {
+		for _, i := range y {
+			if strings.EqualFold(o[0], i[0]) && strings.EqualFold(o[1], i[1]) {
+				count++
+			}
+		}
+	}
+
+	return len(x) == count
+}
+
+func GetSystemAddrPool(client *gophercloud.ServiceClient, addrpool_instance *starlingxv1.AddressPool) (*addresspools.AddressPool, error) {
+	var found_addrpool *addresspools.AddressPool
+	addrpool_list, err := addresspools.ListAddressPools(client)
 	if err != nil {
-		err = perrors.Wrap(err, "failed to list address pools")
-		return err
+		logHost.Error(err, "failed to fetch addresspools from system")
+		return nil, err
+	}
+	// Preferably fetch the addresspool using UUID.
+	if addrpool_instance.Status.ID != nil {
+		found_addrpool = utils.GetSystemAddrPoolByUUID(addrpool_list, *addrpool_instance.Status.ID)
+		if found_addrpool != nil {
+			return found_addrpool, nil
+		}
+
+		// In case addresspool instance is using a stray ID fetch it by name.
+		found_addrpool = utils.GetSystemAddrPoolByName(addrpool_list, addrpool_instance.Name)
+
+	} else {
```

Comment:
I don't think we need this else or the found_addrpool = utils.GetSystemAddrPoolByName(addrpool_list, addrpool_instance.Name) 2x. If found with UUID return, if not return using name.

Reply:
It's updated now, please check